### PR TITLE
Improve mk1 usart

### PR DESCRIPTION
**Pull request by stieg**
Improves the code used for the USART on MK1 by re factoring it and eliminating a bunch of the duplicated code and portions of code with externs and general bad practice coding in place.

### CODE REVIEW

File: SAM7s_base/usart_at91/usart_device_at91.c [hunk 1]
```diff
@@ -142,9 +134,17 @@ static unsigned int createUartMode(unsigned int bits, unsigned int parity, unsig
...
     AT91F_PMC_EnablePeriphClock( AT91C_BASE_PMC,(1 << AT91C_ID_US0) | (1 << AT91C_ID_US1));
-    if (!initQueues()) return 0;
+
+    /* Init device 0 (Telemetry) */
+    if (!init_txrx_queue(0, 512, 512))
+            return 0;
     usart_device_init_0(8, 0, 1, TELEMETRY_BAUDRATE);
+
+    /* Init device 1 (GPS) */
+    if (!init_txrx_queue(1, 512, 128))
```
Comment (brentpicasso) [inline SAM7s_base/usart_at91/usart_device_at91.c]: Rx queue should be larger than Tx queue.  Tx Queue can be minimal, like 16. MK1 does not communicate with GSP module, only receives data.

suggest: init_txrx_queue(1, 16, 512)

Also, extract magic numbers to #defines

File: SAM7s_base/usart_at91/usart_device_at91.c [hunk 2]
```diff
@@ -142,9 +134,17 @@ static unsigned int createUartMode(unsigned int bits, unsigned int parity, unsig
 int usart_device_init()
 {
     AT91F_PMC_EnablePeriphClock( AT91C_BASE_PMC,(1 << AT91C_ID_US0) | (1 << AT91C_ID_US1));
-    if (!initQueues()) return 0;
+
+    /* Init device 0 (Telemetry) */
+    if (!init_txrx_queue(0, 512, 512))
```
Comment (brentpicasso) [inline SAM7s_base/usart_at91/usart_device_at91.c]: extract magic numbers

Comment (stieg) [inline SAM7s_base/usart_at91/usart_device_at91.c]: Reason?  Not against it, just seems pointless unless its part of a larger config picture.

File: SAM7s_base/usart_at91/usart_device_at91.c (hunk 1 above)
Comment (stieg) [inline SAM7s_base/usart_at91/usart_device_at91.c]: Oops... meant this to be the other way around.  Will apply that feedback.

**Comment by stieg** [comment]
Also I forgot to update the CHANGELOG entry to 2.8.6.  Need to fix that too

File: SAM7s_base/usart_at91/usart_device_at91.c (hunk 2 above)
Comment (brentpicasso) [inline SAM7s_base/usart_at91/usart_device_at91.c]: It's a best practice for readability and maintainability. http://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad

Comment (stieg) [inline SAM7s_base/usart_at91/usart_device_at91.c]: Ok.  Will make this part of our coding standard then.

**Comment by brentpicasso** [comment]
This looks good. If you've tested it on MK1 then please :shipit:

**Comment by stieg** [comment]
I'll test before I merge

On Mon, Oct 19, 2015 at 3:46 PM, Brent Picasso <notifications@github.com>
wrote:

> This looks good. If you've tested it on MK1 then please [image: :shipit:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/97#issuecomment-149342257>
> .
>



--

**Comment by stieg** [comment]
Tested and passed after fixing a small typo.  Merging.

**Pull request merged**